packages/vite-plugin-cloudflare/src/plugin-config.ts

```diff
@@ -24,11 +24,16 @@ export interface PluginConfig extends Partial<PluginWorkerConfig> {
 type Defined<T> = Exclude<T, undefined>;
 
 export type AssetsOnlyConfig = SanitizedWorkerConfig & {
+	topLevelName: Defined<SanitizedWorkerConfig['topLevelName']>;
+	name: Defined<SanitizedWorkerConfig['name']>;
+	compatibility_date: Defined<SanitizedWorkerConfig['compatibility_date']>;
 	assets: Defined<SanitizedWorkerConfig['assets']>;
 };
 
 export type WorkerConfig = SanitizedWorkerConfig & {
+	topLevelName: Defined<SanitizedWorkerConfig['topLevelName']>;
 	name: Defined<SanitizedWorkerConfig['name']>;
```

Comment:
could we actually not rely on `name` at all anymore and simply always just use `topLevelName`?

(I'm worried that we'll end up sometimes using `name` and sometimes using `topLevelName` potentially confusing things and even potentially making debugging harder)

Reply:
We do need both. `name` is what is passed to Miniflare and used in the `wrangler.json` output whereas `topLevelName` is what's used as the default environment name (and therefore default output subdirectory).

Reply:
yes, I thought that we could also just pass `topLevelName` to miniflare as well, but that would break service bindings wouldn't it? 😓👍

Reply:
(PS: using those two names is a bit unfortunate to me as I think that it feels a bit messy... but I guess that it can't be helped)